2024/10/14/src/main/java/me/lemire/MyBenchmark.java

```diff
@@ -263,9 +259,11 @@ public static int replaceBackslashRawCompressedTable3(byte[] original, byte[] ne
         int outputOffset = 0;
         for (byte b : original) {
             int latin = b & 0xFF;
-            short replacement = replacementsAndLengthsTables[2 * latin];
+            int replacementAndLength = replacements[latin];
+            short replacement = (short) replacementAndLength;
             SHORT_WRITER.set(newArray, outputOffset, replacement);
-            outputOffset += replacementsAndLengthsTables[(2 * latin) + 1];
+            int length =  replacementAndLength >> 16;
```

Comment:
Can `length` be equals 1? If yes, can we avoid writting as a short? ( I didn't tests that.. )

Pseudo-code
```
if length == 1
  newArray[outputOffset] = (byte) replacement;
else
  SHORT_WRITER.set(newArray, outputOffset, replacement);
```

Reply:
The point of storing it is to save branches to avoid the penalty of a mispredict